docs/design/proposal-troubleshoot-consolidated-cli.md

```diff
@@ -0,0 +1,243 @@
+
+There is discussion about changing the behaviour of `preflight`, considering that preflights and support-bundles utilize the same specs - the same collectors and analyzers - and only differ in what is returned to the user post-analysis.  For this design proposal, `support-bundle` and `preflight` may be condensed into a single `troubleshoot` binary.
+
+For the purposes of this design we'll talk about troubleshoot as a standalone binary. It will also be available, and likley most commonly installed as, a krew plugin for kubectl.
+
+## Non-Goals
+
+## Background
+
+## High Level Design
+
+Functions of `support-bundle`, `preflight`, `analyze`, `redact`, and `sbctl` binaries/tools should be rolled together into a single `troubleshoot` CLI that can perform all necessary functions of the same.
+
+`troubleshoot` CLI should be able to report on the version that is installed in the CLI, and any support bundle generated with `troubleshoot` should report its build/version inside the archive it generates.
+
+## Detailed Design
+
+In the interest of being able to work on this quickly without breaking existing use-cases, a new `troubleshoot` command should be created. Utilizing cobra and viper best practices from the cobra.dev docs.
+
+### sbctl → troubleshoot inspect
+
+sbctl should be migrated to the troubleshoot repository in a "lift and shift" operation to start with.
+
+It should continue to be built as a standalone binary to enable continued use until a stable replacement exists. as such it should me integrated in a way that preserves the existing `cmd` and it's package requirements.
+
+Once integrated into the repository, we can begin using functions from it's packages from the from the new `troubleshoot` command to enable the `inspect` subcommand.
+
+### Public APIs
+
+The existing package entrypoint we use from the support-bundle command is `supportbundle.CollectSupportBundleFromSpec()`
+which is an end-to-end function that runs collectors, redactors and then analysis.
+
+The Consolidated cli should be responsible for handling the end-to-end running of this process in the `cmd` leaving the importable packages free to focus on defining the individual collect, redact analyze steps.
+
+To this end we should create a new `pkg` that targets the functionality provided by the `collect` and `analyze` packages such as `collect.runHostCollectors()` and provide a stable API to be used by the CLI and other projects. At the same time we can unpublish functionality that should not be exposed (such as the individual collector and analyser functions) and mark code for deprecation.
+
+This new package should be kept as minimal as possible. and serve only as an interface to private functions in the other packages.
+
+Once the stable API is ready we can instruct projects like kurl to target that and work on removing code marked for deprecation.
+
+The functionality we want to expose via this api is:
+
+- `CollectBundle(spec{}) (bundlePath string, error)`
```

Comment:
_This is more of an implementation detail._

For the public API function parameters, we should lean towards having a `struct` as a single parameter rather than inlined parameters.

Preference
```go
type CollectOptions struct {
	SpecURIs []string
	AnotherArg string
	// Future args come here
}

func CollectBundle(opt CollectOptions) (bundlePath string, error)
```

instead of 
```go
func CollectBundle(specURIs string[], anotherArg string) (bundlePath string, error)
```

With this approach the function signature remains constant but we are able to pass in new parameters over time, future proofing the API.